src/plugin/indexer-elastic/src/java/org/apache/nutch/indexwriter/elastic/ElasticIndexWriter.java

```diff
@@ -162,6 +191,17 @@ protected RestHighLevelClient makeClient(IndexWriterParams parameters) throws IO
       } else	{
         LOG.debug("No cluster name provided so using default");
       }
+      //Only add username and password if password is configured
+      if(StringUtils.isNotBlank(password)) {
```

Comment:
Great, @balashashanka! Looks good and should work.

Minor point: If possible try to apply the [Nutch Eclipse Code Formatting rules](https://github.com/apache/nutch/blob/master/eclipse-codeformat.xml) and also remove trailing white space. Are you using Eclipse? If not let me know or whether you need help to set it up.

Reply:
Hi @sebastian-nagel, did the formatting.